backend/prisma/userSeed.ts

```diff
@@ -0,0 +1,141 @@
+import prisma from '../utils/prismaClient'
+import { RelationshipStatus } from '@prisma/client'
+import crypto from 'crypto'
+const db = prisma
+
+const realUsers = [
+  {
+    email: 'test1@gmail.com',
+    firstName: 'john',
+    lastName: 'smith'
+  }, {
+    email: 'test2@gmail.com',
+    firstName: 'Xavier',
+    lastName: 'Moss'
+  }, {
+    email: 'test3@gmail.com',
+    firstName: 'Kaylee',
+    lastName: 'Ryan'
+  }, {
+    email: 'test4@gmail.com',
+    firstName: 'Scott',
+    lastName: 'Lee'
+  }, {
+    email: 'test5@gmail.com',
+    firstName: 'Carly',
+    lastName: 'Hendrix'
+  }, {
+    email: 'test6@gmail.com',
+    firstName: 'Ella',
+    lastName: 'Haas'
+  }, {
+    email: 'test7@gmail.com',
+    firstName: 'Shania',
+    lastName: 'Baxter'
+  }, {
+    email: 'test8@gmail.com',
+    firstName: 'Halle',
+    lastName: 'Pratt'
+  }, {
+    email: 'test9@gmail.com',
+    firstName: 'Miya',
+    lastName: 'Reid'
+  }, {
+    email: 'test10@gmail.com',
+    firstName: 'Wesley',
+    lastName: 'Lowery'
+  }, {
+    email: 'test111@gmail.com',
+    firstName: 'Bianca',
+    lastName: 'Lawrence'
+  }
+]
+
+const testUser = {
+  email: 'test@gmail.com',
+  firstName: 'bob',
+  lastName: 'admin'
+}
+
+const main = async (): Promise<void> => {
+  // create a test user,
+  try {
+    const salt = crypto.randomBytes(16).toString('base64');
+    crypto.pbkdf2('test', salt, 310000, 32, 'sha256', async function (err, hashedPassword) {
+      if (err !== null) throw new Error('failed to create test user')
+
+      await db.user.create({
+        data: {
+          email: testUser.email,
+          firstName: testUser.firstName,
+          lastName: testUser.lastName,
+          password: hashedPassword.toString('base64'),
+          salt
+        }
+      })
+    })
+  } catch (err) {
+    console.error(err)
+  }
+
+  // Execute 1 sec later
+  setTimeout(async () => {
+    const bobTheTestUser = await db.user.findFirstOrThrow({
+      where: {
+        firstName: testUser.firstName
+      }
+    })
+    // create many users
+    await db.user.createMany({
+      data: realUsers
+    })
+
+    // bob adds a bunch of user
+    realUsers.map(async user => {
```

Comment:
part of the problem here too is that the .map call is not asynchronous, so it is not going to pause execution of js on this entire .map call finishing. it will queue up all of these async fn's and they will just kind of finish whenever they happen to finish. you could try looking into using Promise.all here

Reply:
Fixed this with promise.all()